os/drivers/input/ist415.c

```diff
@@ -1270,14 +1269,26 @@ int ist415_initialize(const char *path, struct i2c_dev_s *i2c, struct ist415_con
 	upper = (struct touchscreen_s *)kmm_zalloc(sizeof(struct touchscreen_s));
 	if (!upper) {
 		ist415dbg("Fail to alloc touchscreen_s\n");
-		kmm_free(dev);
 		ist415_disable(dev);
-		return ERROR;
+		goto cleanup_task_wd;
 	}
 	upper->ops = &g_ist415_ops;
 	upper->priv = dev;
 	dev->upper = upper;
+	ret = touch_register(path, upper);
+	if(ret < 0) {
+		ist415_disable(dev);
+		goto cleanup_task_wd;
+	}
+	return ret;
 
-	ist415dbg("Touch Driver registered Successfully\n");
-	return touch_register(path, upper);
+/* Error Handling Releasing all allocated resources */
+cleanup_task_wd:
+	wd_delete(dev->wdog);
+	task_delete(dev->pid);
+cleanup_sems_mem:
+	sem_destroy(&dev->sem);
+	sem_destroy(&dev->wait_irq);
+	kmm_free(dev);
```

Comment:
The touch driver registering fail is critical issue.
So, How about add DEBUGASSERT here for detect early?